src/ucp/api/ucp.h

```diff
@@ -107,6 +108,52 @@ BEGIN_C_DECLS
  */
 
 
+/**
+ * For internal use only.
+ */
+extern size_t ucp_request_internal_size;
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief A helper macro to allocate request on stack.
+ *
+ * A helper macro to allocate request on stack according to required memory
+ * layout to use @ref UCP_REQUEST_FLAG_EXTERNAL. 
+ *
+ * @note @ref UCP_REQUEST_FLAG_EXTERNAL also should be added to
+ * @ref ucp_request_t::flags initialization to hint UCP library.
+ *
+ * @note Useful for implementation of blocking operations like MPI_Send/MPI_Recv.
+ *
+ * @param [in]  _request_size   User request size, e.g. equal to
+ *                              @ref ucp_context_attr_t::requst_size.
+ *
+ * @return Pointer to ucp_request_attr_t for initialization.
+ */
+#define UCP_REQUEST_ALLOCA(_request_size) \
```

Comment:
what is ALLOCA ? Allocator ?

Reply:
just a helper macro around `alloca` returning correct pointer to built-in attrs inside "external" request allocated on stack

Reply:
Could we please choose a solution that works for languages that will wrap UCX that don't have `alloca`? ie, all of them...

Reply:
this macro is not mandatory, a user can allocate the request in any desirable way